### PR TITLE
fix code for computing least sort of KItems

### PR DESCRIPTION
**Pull request by dwightguth**
I fully expect removing this nasty hack from Context.java will break things. So don't review until you see the tests pass.

### CODE REVIEW

**Comment by dwightguth** [comment]
@yilongli please review.

**Comment by radumereuta** [comment]
@dwightguth @yilongli  you should have a look at #633 first. I think there is a similar problem there.

File: src/javasources/KTool/src/org/kframework/kil/loader/Context.java [hunk 1]
```diff
@@ -190,7 +190,6 @@ public void putLabel(Production p, String cons) {
 //            labels.put(label, s = new HashSet<String>());
 //        }
 //        s.add(cons);
-        putLabel(p.getLabel(), cons);
```
Comment (yilongli) [inline src/javasources/KTool/src/org/kframework/kil/loader/Context.java:193]: Is this change related to this fix?

Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/kil/loader/Context.java:193]: Yes. Previously if you had two productions whose klabel differed only in
the initial apostrophe, both productions would be considered as being
associated with the klabel without the apostrophe. That is because of this
line of code, which is a bug.

**Comment by yilongli** [comment]
lgtm
